### PR TITLE
Set up basic GUI for rules

### PR DESCRIPTION
**Pull request by tcgj**
- Completes #39

### CODE REVIEW

File: src/main/java/budgetbuddy/model/rule/expression/ActionExpression.java [hunk 1]
```diff
@@ -48,6 +48,11 @@ public Value getValue() {
         return value;
     }
 
+    @Override
+    public String getType() {
+        return RuleAction.TYPE_EXPRESSION;
+    }
+
```
Comment (kenneth-fung) [inline src/main/java/budgetbuddy/model/rule/expression/ActionExpression.java]: Why not have `getType()` in `RuleAction` return `TYPE_EXPRESSION` ? Then you don't need to override for each child class. Unless a future child class might return something different. Same for `RulePredicate` .

Comment (tcgj) [reply src/main/java/budgetbuddy/model/rule/expression/ActionExpression.java]: Yup, this is in consideration of possible scripts that are also ruleactions/rulepredicates

Comment (kenneth-fung) [reply src/main/java/budgetbuddy/model/rule/expression/ActionExpression.java]: Huh, sorry I'm not understanding. If a script extends `RuleAction`, then wouldn't it inherit `getType()` and its implementation?

Comment (tcgj) [reply src/main/java/budgetbuddy/model/rule/expression/ActionExpression.java]: So getType() is in both RulePredicate and RuleAction as abstract methods. so when a script/expression extends it, it'll have to implement getType, but perhaps with a different type. Hence only ActionExpression has TYPE_EXPRESSION.

Comment (kenneth-fung) [reply src/main/java/budgetbuddy/model/rule/expression/ActionExpression.java]: Oh hm ok, you're probably more aware of what's extending what so I'll leave it to you.

**Pull request merged**